awesome.c

```diff
@@ -901,20 +913,44 @@ main(int argc, char **argv)
 
     ewmh_init_lua();
 
+    /* Parse and run configuration file before adding the screens */
+    if (globalconf.no_auto_screen)
+    {
+        /* Disable automatic screen creation, awful.screen has a fallback */
+        globalconf.ignore_screens = true;
+
+        if(!luaA_parserc(&xdg, confpath))
+            fatal("couldn't find any rc file");
+    }
+
+    screen_emit_scanning();
+
     /* init screens information */
     screen_scan();
 
-    /* Parse and run configuration file */
-    if (!luaA_parserc(&xdg, confpath))
+    /* Parse and run configuration file after adding the screens */
+    if (((!globalconf.no_auto_screen) && !luaA_parserc(&xdg, confpath)))
         fatal("couldn't find any rc file");
 
     p_delete(&confpath);
 
     xdgWipeHandle(&xdg);
 
+    screen_emit_scanned();
+
+    /* Exit if the user doesn't read the instructions properly */
+    if (globalconf.no_auto_screen && !globalconf.screens.len)
+        fatal("When -m/--screen is set to \"off\", you **must** create a "
+              "screen object before or inside the screen \"scanned\" "
+              " signal. Using AwesomeWM with no screen is **not supported**.");
+
+    client_emit_scanning();
+
     /* scan existing windows */
     scan(tree_c);
 
+    client_emit_scanned();
```

Comment:
Should these signal emissions be moved to `scan` (and same for the screen scanning signals being moved to `screen_scan()`)? It feels to me like they belong there.

Reply:
done